pkg/controllers/hub/membercluster/controller.go

```diff
@@ -0,0 +1,126 @@
+package membercluster
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	fleetnetv1alpha1 "go.goms.io/fleet-networking/api/v1alpha1"
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+)
+
+var (
+	fleetMemberNamespace = "fleet-member-%s"
+)
+
+const (
+	ControllerName = "membercluster-watcher"
+)
+
+// Reconciler reconciles a MemberCluster object.
+type Reconciler struct {
+	client.Client
+	Recorder record.EventRecorder
+	// the wait time in minutes before we need to force delete a member cluster.
+	ForceDeleteWaitTime time.Duration
+}
+
+// Reconcile watches the deletion of the member cluster and removes finalizers on fleet networking resources in the
+// member cluster namespace.
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	mcObjRef := klog.KRef(req.Namespace, req.Name)
+	startTime := time.Now()
+	klog.V(2).InfoS("Reconciliation starts", "memberCluster", mcObjRef)
+	defer func() {
+		latency := time.Since(startTime).Milliseconds()
+		klog.V(2).InfoS("Reconciliation ends", "memberCluster", mcObjRef, "latency", latency)
+	}()
+	var mc clusterv1beta1.MemberCluster
+	if err := r.Client.Get(ctx, req.NamespacedName, &mc); err != nil {
+		if errors.IsNotFound(err) {
+			klog.V(4).InfoS("Ignoring NotFound memberCluster", "memberCluster", mcObjRef)
+			return ctrl.Result{}, nil
+		}
+		klog.ErrorS(err, "Failed to get memberCluster", "memberCluster", mcObjRef)
+		return ctrl.Result{}, err
+	}
+	// Handle deleting member cluster, removes finalizers on all the resources in the cluster namespace
+	// after member cluster force delete wait time.
+	if !mc.DeletionTimestamp.IsZero() && time.Since(mc.DeletionTimestamp.Time) >= r.ForceDeleteWaitTime {
+		klog.V(2).InfoS("The member cluster deletion is stuck, "+
+			"garbage collect all the resources in member cluster namespace", "memberCluster", mcObjRef)
```

Comment:
```suggestion
			"removing the finalizers from all the resources in member cluster namespace", "memberCluster", mcObjRef)
```
The garbage collect is actually done by the hub agent in the fleet repo.